torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -491,6 +492,8 @@ def __init__(
         _init_runtime_state(self)
         _init_prefetching_state(self, backward_prefetch, forward_prefetch)
         _init_buffer_state(self, module)
+        # extension needs to be set before `_init_param_handle_from_module()`
+        _init_extension(self, device_mesh)
```

Comment:
For completeness, shall we include this change for `fully_shard` path too (even though it is not too important right now)?
https://github.com/pytorch/pytorch/blob/c2f5d4d8f09b535e434313223434009bad282258/torch/distributed/_composable/fully_shard.py#L106-L109

Reply:
> For completeness, shall we include this change for `fully_shard` path too (even though it is not too important right now)?
> 
> https://github.com/pytorch/pytorch/blob/c2f5d4d8f09b535e434313223434009bad282258/torch/distributed/_composable/fully_shard.py#L106-L109

Yes. Good point! We are planning to do the composable API changes in follow up PRs.